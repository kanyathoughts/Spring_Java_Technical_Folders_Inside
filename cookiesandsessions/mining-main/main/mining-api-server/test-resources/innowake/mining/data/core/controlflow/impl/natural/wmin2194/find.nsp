DEFINE DATA
LOCAL
1 #VAR1 (P11.2)
1 EMP VIEW OF EMPLOYEES
 2 PERSONNEL-ID
 2 JOB-TITLE
 2 FIRST-NAME
END-DEFINE

FIND
	EMP PERSONNEL-ID GT '30000311'
	WRITE NOTITLE '0'
END-FIND

FIND EMP WITH PERSONNEL-ID GT '40000311' SORTED BY PERSONNEL-ID
  WHERE *TRANSLATE(JOB-TITLE, UPPER) = 'INGENIEUR COMMERCIAL'
    WRITE NOTITLE '1'
END-FIND

FIND
	(5)
		EMP
			WITH
				PERSONNEL-ID GT '40000311'
					SORTED BY
						PERSONNEL-ID
  WHERE *TRANSLATE(JOB-TITLE, UPPER) = 'INGENIEUR COMMERCIAL'
    WRITE NOTITLE '2'
END-FIND

FIND
	(#VAR1)
		IN
			EMP
				WITH
					PERSONNEL-ID GT '40000311' AND 
						JOB-TITLE EQ 'IT' OR
						JOB-TITLE EQ 'IT2'
  WHERE *TRANSLATE(JOB-TITLE, UPPER) = 'INGENIEUR COMMERCIAL'
    WRITE NOTITLE '3'
END-FIND

FIND FIRST EMP WITH PERSONNEL-ID GT '50000311'
	WHERE *TRANSLATE(JOB-TITLE, UPPER) = 'IT Guy'
    	WRITE NOTITLE '4'
END-FIND

FIND UNIQUE EMP WITH PERSONNEL-ID GT '60000311'
	WHERE *TRANSLATE(JOB-TITLE, UPPER) = 'IT Guy2'
    	WRITE NOTITLE '5'
END-FIND

FIND (10) EMP FIRST-NAME NE 'MORENO' STARTING WITH ISN = #VAR1 SORTED BY FIRST-NAME
  WRITE NOTITLE '6'
END-FIND

END