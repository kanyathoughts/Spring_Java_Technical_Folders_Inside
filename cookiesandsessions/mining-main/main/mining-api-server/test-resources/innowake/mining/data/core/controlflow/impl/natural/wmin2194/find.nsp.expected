 #2 DefineDataStmt
 #12 FindStmt, Label: FIND EMP PERSONNEL-ID GT '30000311', FROM={#18, TestEntryPoint}, TO={#18(LOOP BODY), #22(LOOP END)}
  #18 WRITE NOTITLE '0', FROM={#12(LOOP BODY)}, TO={#12}
 #22 FindStmt, Label: FIND EMP WITH PERSONNEL-ID GT '40000311' SORTED BY PERSONNEL-ID WHERE *TRANSLATE(JOB-TITLE, UPPER) = 'INGENIEUR COMMERCIAL', FROM={#12(LOOP END), #37}, TO={#37(LOOP BODY), #41(LOOP END)}
  #37 WRITE NOTITLE '1', FROM={#22(LOOP BODY)}, TO={#22}
 #41 FindStmt, Label: FIND (5) EMP WITH PERSONNEL-ID GT '40000311' SORTED BY PERSONNEL-ID WHERE *TRANSLATE(JOB-TITLE, UPPER) = 'INGENIEUR COMMERCIAL', FROM={#22(LOOP END), #57}, TO={#57(LOOP BODY), #61(LOOP END)}
  #57 WRITE NOTITLE '2', FROM={#41(LOOP BODY)}, TO={#41}
 #61 FindStmt, Label: FIND (#VAR1) IN EMP WITH PERSONNEL-ID GT '40000311' AND JOB-TITLE EQ 'IT' OR JOB-TITLE EQ 'IT2' WHERE *TRANSLATE(JOB-TITLE, UPPER) = 'INGENIEUR COMMERCIAL', FROM={#41(LOOP END), #87}, TO={#87(LOOP BODY), #91(LOOP END)}
  #87 WRITE NOTITLE '3', FROM={#61(LOOP BODY)}, TO={#61}
 #91 FindStmt, Label: FIND FIRST EMP WITH PERSONNEL-ID GT '50000311' WHERE *TRANSLATE(JOB-TITLE, UPPER) = 'IT Guy', FROM={#105, #61(LOOP END)}, TO={#105(LOOP BODY), #109(LOOP END)}
  #105 WRITE NOTITLE '4', FROM={#91(LOOP BODY)}, TO={#91}
 #109 FindStmt, Label: FIND UNIQUE EMP WITH PERSONNEL-ID GT '60000311' WHERE *TRANSLATE(JOB-TITLE, UPPER) = 'IT Guy2', FROM={#123, #91(LOOP END)}, TO={#123(LOOP BODY), #127(LOOP END)}
  #123 WRITE NOTITLE '5', FROM={#109(LOOP BODY)}, TO={#109}
 #127 FindStmt, Label: FIND (10) EMP FIRST-NAME NE 'MORENO' STARTING WITH ISN = #VAR1 SORTED BY FIRST-NAME, FROM={#109(LOOP END), #137}, TO={#137(LOOP BODY), #141(LOOP END)}
  #137 WRITE NOTITLE '6', FROM={#127(LOOP BODY)}, TO={#127}
 #141 EndStmt, FROM={#127(LOOP END)}, TO={TestReturnPoint}
